src/env.cc

```diff
@@ -457,7 +457,7 @@ void Environment::RunAndClearNativeImmediates() {
 void Environment::CheckImmediate(uv_check_t* handle) {
   Environment* env = Environment::from_immediate_check_handle(handle);
 
-  if (env->immediate_info()->count() == 0)
+  if (env->immediate_info()->count() == 0 || !env->can_call_into_js())
```

Comment:
I think this should be right before the `do {`, because we probably want our native immediate callbacks to still be called – not all of them are calling JS, some or doing cleanup rather than something else

Reply:
Would that be strictly correct even if it's not calling JS? At least in relation to the main thread (and not workers), the expectation was that no user code would run after `EmitExit` finishes. After `RunCleanup` was added that's no longer the case since we run the loop one final time.

Are there are any Immediates that we truly want to have run at this point?